web/crux-ui/src/hooks/use-persisted-view-mode.ts

```diff
@@ -0,0 +1,28 @@
+import { ViewMode } from '@app/components/shared/view-mode-toggle'
+import { useEffect, useState } from 'react'
+
+function usePersistedViewMode({
+  initialViewMode,
+  pageName,
+}: {
+  initialViewMode: ViewMode
+  pageName: string
+}): [ViewMode, (mode: ViewMode) => void] {
+  const [viewMode, setViewMode] = useState<ViewMode>(() => {
+    if (typeof window !== 'undefined') {
+      const storedViewMode = localStorage.getItem(`viewMode_${pageName}`) as ViewMode
+      return storedViewMode || initialViewMode
+    }
+    return initialViewMode
+  })
+
+  useEffect(() => {
```

Comment:
You could also just return a custom setViewMode function which calls setState and localStorage.setItem, but that's just nitpicking :)

Otherwise tested, looks good 🚀